sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -93,90 +93,56 @@ def generate_pip_requirements(self, target_file):
     """ write the python packages to a requirement file
     the generated file follows the order of which the packages are added """
```

Comment:
the new order is by the package name.

Reply:
Correct. Previously it is a dictionary, which does not reserve the orders.

Reply:
Nice catch. I used the OrderedDict. Revert the change to preserve the orders.